Bottom: 6d0312f81e9cbef8df333c70e9dd67053234fd79
Top:    7f079afa759ea43016f74655a9d688421dd31cbf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-22 10:30:43 +0000

Refresh of add-leaky-free-function

---

diff --git a/src/main.c b/src/main.c
index 462285d..735fde7 100644
--- a/src/main.c
+++ b/src/main.c
@@ -53,6 +53,7 @@
 #include "action.h"
 #include "misc.h"
 #include "lrealpath.h"
+#include "free_atexit.h"
 
 #include "hid/common/actions.h"
 
@@ -1028,6 +1029,13 @@ main (int argc, char *argv[])
   /*    FIX_ME
      LoadBackgroundImage (Settings.BackgroundImage); */
 
+  /* This must be called before any other atexit functions
+   * are registered, as it configures an atexit function to
+   * clean up and free various items of allocated memory,
+   * and must be the last last atexit function to run.
+   */
+  leaky_init ();
+
   /* Register a function to be called when the program terminates.
    * This makes sure that data is saved even if LEX/YACC routines
    * abort the program.
